metricbeat/mb/module/wrapper.go

```diff
@@ -311,6 +310,31 @@ func (msw *metricSetWrapper) Test(d testing.Driver) {
 	})
 }
 
+func (msw *metricSetWrapper) handleFetchError(err error, reporter mb.PushReporterV2) {
+	switch {
+
+	case errors.As(err, &mb.PartialMetricsError{}):
+		reporter.Error(err)
+		msw.consecutiveErrors = 0
+		// mark module as running if metrics are partially available and display the error message
+		msw.module.UpdateStatus(status.Running, fmt.Sprintf("Error fetching data for metricset %s.%s: %v", msw.module.Name(), msw.MetricSet.Name(), err))
+		logp.Err("Error fetching data for metricset %s.%s: %s", msw.module.Name(), msw.Name(), err)
+
+	case err != nil:
+		reporter.Error(err)
+		msw.consecutiveErrors++
```

Comment:
nit: I am not an expert in MetricBeat/MetricSets code but if `fetch` is called by multiple goroutines here we should go with atomics?! If it shouldn't be called like that maybe we should enrich the fetch godoc to mark it explicitly as a concurrent unsafe

Reply:
My understanding from [here](https://github.com/elastic/beats/blob/ce4a17b3a960f779d61a6898f97080bbd2423033/metricbeat/mb/module/wrapper.go#L135-L154) is that each metricset is run within its own goroutine so I didn't add any extra synchronization around the counters.
@leehinman can maybe have a look and if there's the chance of a race condition we can easily switch to atomics

Reply:
@leehinman I had a look at using a *monitoring.Int (which is just a struct wrapping an atomic.Int64) and I was quite surprised to find some of the new unit tests failing.
I then realized that `stats` keeps state from previous tests thanks to [this](https://github.com/elastic/beats/blob/b32f635c7edfbaaa3a2aa68dcb4698da163c025a/metricbeat/mb/module/wrapper.go#L53) and the [getMetricSetStats()](https://github.com/elastic/beats/blob/b32f635c7edfbaaa3a2aa68dcb4698da163c025a/metricbeat/mb/module/wrapper.go#L451-L473) function.

Is it normal that we keep state from a previous wrapper instance (in the unit tests the metricsetWrapper is recreated along with the mocks for each testcase) just matching on metricset name ? I am not sure we want to do the same for the consecutive errors part just because a previous wrapper existed that failed all the time...
Any thoughts ? Is remembering previous states for previous wrappers of the metricset the standard behavior in metricbeat ?